### DISCUSSION TITLE
Unable to calculate taxes - We are not able to calculate taxes for your order.  [AVALARA PLUGIN]

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#8129] [General]
Hi guys,
I've retrieve this general error using **Saleor 3.1.0-a.0.**, **SDK 0.3.0-10** and **Avalara plugin**:

```code: 'TAX_ERROR', field: null, message: 'Unable to calculate taxes - We are not able to calculate taxes for your order. Please try later', __typename: 'CheckoutError'```

Avalara plugin is active.
We removed the flag `All products prices are entered with tax included` in _Configuration -> Taxes_ to charge the amount on the final price.

Do you have any advice to have more detail about the issue?

Thanks ina dvance and kind regards,

Davide

**Comment by korycins** [comment]
@DidoMarchet Are you sure that your credentials are correct?
You can confirm them by calling: 
Prod endpoint
https://rest.avatax.com/api/v2/utilities/ping - with basic auth: (username_or_account, password_or_licence) 
Sandbox endpoint
https://sandbox-rest.avatax.com/api/v2/utilities/ping - with basic auth: (username_or_account, password_or_licence)

Make sure that you provided the correct address in dashboard/configuration/plugins/avatax.
Make sure that you set up `Use Sandbox` switch in a proper way. 

You can also check the backend logs. If you received such an error, then there should be a warning with an error msg from Avatax.

**Reply by DidoMarchet** [reply]
Hi!
Thanks for the answer.
We'll check as soon as possible and if it's solved I mark as answer:

Have a nice day and kind regards,

Davide

**Reply by DidoMarchet** [reply]
@korycins Hi!
We tested the endpoint with a production ready account of Avalara and a sandobx account.
Both the calls returned the right value.

We try to configure the plugin in Saleor 3.1.0-a.0. and we seen the **production account works as aspected** instead **the sandobox account doesn't* (with the sandbox option flagged into the plugin configuration).

Do you have any advice about ?

Thanks in advance and kind regards,

Davide

**Reply by korycins** [reply]
Hey @DidoMarchet ,
This is strange, we use Avatax a lot and we don't have such issues. Do you have any logs with responses from Avatax? It should be visible as a warning. 
The one thing which could be helpful in debugging is, logging a request payload and response payload from Avatax.

**Reply by DidoMarchet** [reply]
Hi,
thanks again for the answer.
We'll check!

Have a nice ay and regards,

Davide
